marlowe-playground-client/src/Marlowe/ActusBlockly.purs

```diff
@@ -0,0 +1,742 @@
+  terminationDate <- fromMaybe maturityDate <$> actusDateToDay c.terminationDate
+  purchaseDate <- fromMaybe maturityDate <$> actusDateToDay c.purchaseDate
+  rateResetCycle <- blocklyCycleToCycle c.rateReset
+  rateResetAnchorValue <- blocklyCycleToAnchor c.rateReset
+  rateResetAnchor <- sequence $ actusDateToDay <$> rateResetAnchorValue 
+  notional <- Either.note "notional is a mandatory field!" <$> actusDecimalToNumber c.notional >>= identity
+  premium <- fromMaybe 0.0 <$> actusDecimalToNumber c.premiumDiscount
+  interestRateCycle <- blocklyCycleToCycle c.interestRateCycle
+  interestRateAnchorValue <- blocklyCycleToAnchor c.interestRateCycle
+  interestRateAnchor <- sequence $ actusDateToDay <$> interestRateAnchorValue
+  interestRateUnchecked <- actusDecimalToNumber c.interestRate
+  interestRate <- if isJust interestRateCycle && isNothing interestRateUnchecked 
+          then Either.Left "Please specify interest rate"
+          else Either.Right interestRateUnchecked 
+  assertionCtx <- blocklyAssertionCtxToAssertionCtx c.assertionCtx
+  assertion <- blocklyAssertionToAssertion c.assertion
+  let constraint ctx = Assertions {
+    context: ctx
+    , assertions : (case assertion of
+        Just x -> [x]
+        Nothing -> [])
+  }
+
+  pure $ ContractTerms
+      { contractId : "0"
+      , contractType : PAM
+      , ct_IED : initialExchangeDate
+      , ct_SD : startDate
+      , ct_MD : maturityDate
+      , ct_TD : terminationDate
+      , ct_PRD : purchaseDate
+      , ct_CNTRL : CR_ST
+      , ct_PDIED : premium
+      , ct_NT : notional
+      , ct_PPRD : 0.0
+      , ct_PTD : 0.0
+      , ct_DCC : DCC_A_360
+      , ct_PREF : PREF_N
+      , ct_PRF : CS_PF
+      , scfg : ScheduleConfig {
+            calendar : []
+            , includeEndDay : true
+            , eomc : EOMC_EOM
+            , bdc : BDC_NULL
+        }
+      , ct_PYRT : 0.0
+      , ct_PYTP : PYTP_A
+      , ct_cPYRT : 0.0
+      , ct_OPCL : Nothing
+      , ct_OPANX : Nothing
+      , ct_SCIED : 0.0
+      , ct_SCEF : SE_000
+      , ct_SCCL : Nothing
+      , ct_SCANX : Nothing
+      , ct_SCIXSD : 0.0
+      , ct_RRCL : rateResetCycle
+      , ct_RRANX : rateResetAnchor >>= identity
+      , ct_RRNXT : Nothing
+      , ct_RRSP : 0.0
+      , ct_RRMLT : 1.0
+      , ct_RRPF : -1.0
+      , ct_RRPC : 1.0
+      , ct_RRLC : 1.0
+      , ct_RRLF : 0.0
+      , ct_IPCED : Nothing
+      , ct_IPCL : interestRateCycle
+      , ct_IPANX : interestRateAnchor >>= identity
+      , ct_IPNR : interestRate
+      , ct_IPAC : Nothing
+      , ct_FECL : Nothing
+      , ct_FEANX : Nothing
+      , ct_FEAC : Nothing
+      , ct_FEB : FEB_N
+      , ct_FER : 0.0
+      , ct_CURS : false
+      , constraints : constraint <$> assertionCtx
+      }
+
+
+
+aesonCompatibleOptions :: Options
+aesonCompatibleOptions =
+  defaultOptions
+    { unwrapSingleConstructors = true
+    , sumEncoding = aesonSumEncoding
+    }
```

Comment:
isn' this defined somewhere else?

Reply:
I use specific options here - as this JSON coding/decoding is for internal purposes of ACTUS Blockly.